numpy/ma/mrecords.py

```diff
@@ -221,7 +221,8 @@ def __getattribute__(self, attr):
             except IndexError:
                 # Couldn't find a mask: use the default (nomask)
                 pass
-            hasmasked = _mask.view((bool, (len(_mask.dtype) or 1))).any()
+            tp_len = len(_mask.dtype)
+            hasmasked = _mask.view((bool, ((tp_len,) if tp_len else ()))).any()
```

Comment:
This if condition should be `_mask.dtype.fields is None`, else it blows up on structured types with 0 fields. Arguably out of scope, but may as well fix this.

If you're looking for another deprecation for another PR, len(dtype) is looking like a bad idea on non structured dtypes...

Reply:
Although actually the function is crap anyway and should be using one of the recursive helpers in ma/core that ors together all the fields

Reply:
So should I update this PR, or just leave it as it is and let you do a followup?  (I would prefer the latter :))

Reply:
I suppose you could argue that this PR is about fixing things for length-1 subarrays, and you've succeeded even without fixing this. I think fine to put in as is, assuming we can sort out the ci failure by restarting